src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py

```diff
@@ -131,6 +130,14 @@ def get_vm_size_completion_list(prefix, action, parsed_args, **kwargs):#pylint:
 register_cli_argument('vmss extension image', 'top', help='Return top number of records')
 register_cli_argument('vmss extension image', 'version', help='Extension version')
 
+for scope in ['vm diagnostics', 'vmss diagnostics']:
+    register_cli_argument(scope, 'is_windows_os', action='store_true', help='VM uses Windows OS')
+    register_cli_argument(scope, 'settings', help='json string or a file path, which defines data to be collected.')
```

Comment:
Per discussion, the promotion will start 
1. We all agreed on the promotion. Currently cli works with file path value even w/o `@`, arguably is better :-).
2. Then we update all existing usages. I quickly counted it and saw we have roughly 10 places.
3. Document this trick, and make sure people get to know it easily